Dockerfile

```diff
@@ -7,7 +7,7 @@
 # Every target has a BIN_NAME argument that must be provided via --build-arg=BIN_NAME=<name>
 # when building.
 
-ARG GO_VERSION=1.23.3
+ARG GO_VERSION=1.24
```

Comment:
@sheneska, could you please change versions to `1.24.1`(it was just released)? Here and in other places too. There is one more place to bump the Go version `charts/test/go.mod`.